### PR TITLE
Convert useGetGames hook to TypeScript

### PR DESCRIPTION
**Pull request by wariofdz**
### Changed:

- Added `@cornie-js/api-models` as workspace in package.json.
- Added extensions to files .ts and .tsx in jest.config.cjs.
- Added new types `UseGetGamesResult` and `UseGetGamesState`.
- Transformed `useGetGames` hook and his test from JS to TS.
- Transformed `HomeWithAuth` component and his test from JSX to TSX.
- Transformed `GameList` component and his test from JSX to TSX.
- Transformed `GameListItem` component and his test from JSX to TSX.
- Added new type `InitialStateToken`.
- Transformed `createInitialState` function and his test from JS to TS.
- Transformed `authSliceStatus` constants from JS to TS.

### CODE REVIEW

File: packages/frontend/web-ui/src/app/store/helpers/createInitialState.spec.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import { createInitialState } from './createInitialState';
+import { InitialStateToken } from './models/InitialStateToken';
 import {
   STATUS_AUTH_AUTHENTICATED,
   STATUS_AUTH_NOT_AUTHENTICATED,
 } from '../data/authSliceStatus';
 
 describe(createInitialState.name, () => {
-  let getItemSpy;
+  let getItemSpy: any;
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/helpers/createInitialState.spec.ts:10]: Use a better type. Jest has types for this

File: packages/frontend/web-ui/src/app/store/helpers/models/InitialStateToken.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export interface InitialStateToken {
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/helpers/models/InitialStateToken.ts:1]: This is the auth state, right? Should it be `AuthState`?

**Review by notaphplover — CHANGES_REQUESTED** [review]
It looks great, we're close to merge it

File: packages/frontend/web-ui/src/app/store/helpers/createInitialState.spec.ts (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/helpers/createInitialState.spec.ts]: Done!

File: packages/frontend/web-ui/src/app/store/helpers/createInitialState.spec.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 import { createInitialState } from './createInitialState';
+import { AuthState } from './models/AuthState';
 import {
   STATUS_AUTH_AUTHENTICATED,
   STATUS_AUTH_NOT_AUTHENTICATED,
 } from '../data/authSliceStatus';
 
 describe(createInitialState.name, () => {
-  let getItemSpy;
+  let getItemSpy: jest.SpiedFunction<any>;
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/helpers/createInitialState.spec.ts:10]: For now, let's avoid using `any`. Let's use a better type instead:

```suggestion
  let getItemSpy: jest.SpiedFunction<typeof window.localStorage.getItem>;
```

**Review by notaphplover — CHANGES_REQUESTED** [review]
Let's avoid using `any`.

File: packages/frontend/web-ui/src/app/store/helpers/createInitialState.spec.ts (hunk 3 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/helpers/createInitialState.spec.ts:10]: I didn't remember the key word `typeof` but now, I understand completely!

**Review by notaphplover — APPROVED** [review]

**Pull request merged**